src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -59,6 +61,123 @@ def get_server_location(server_name, resource_group_name):
 ###############################################
 
 
+class ClientType(Enum):
+    ado_net = 'ado.net'
+    sqlcmd = 'sqlcmd'
+    jdbc = 'jdbc'
+    php_pdo = 'php_pdo'
+    php = 'php'
+    odbc = 'odbc'
+
+
+class ClientAuthenticationType(Enum):
+    sql_password = 'SqlPassword'
+    active_directory_password = 'ADPassword'
+    active_directory_integrated = 'ADIntegrated'
+
+
+def _get_server_dns_suffx():
+    # Allow dns suffix to be overridden by environment variable for testing purposes
+    from os import getenv
+    return getenv('_AZURE_CLI_SQL_DNS_SUFFIX', default=get_active_cloud().suffixes.sql_server_hostname)
+
+
+def db_show_conn_str(
+        client_provider,
+        database_name='databasename',
+        server_name='servername',
+        auth_type=ClientAuthenticationType.sql_password.value):
+
+    server_suffix = _get_server_dns_suffx()
+
+    conn_str_props = {
+        'server': server_name,
+        'server_fqdn': '{}{}'.format(server_name, server_suffix),
+        'server_suffix': server_suffix,
+        'db': database_name
+    }
+
+    formats = {
+        ClientType.ado_net.value: {
+            ClientAuthenticationType.sql_password.value:
+                'Server=tcp:{server_fqdn},1433;Database={db};User ID=username;'
+                'Password=password;Encrypt=true;Connection Timeout=30;',
```

Comment:
User experience feedback: I would replace the `password` with `<password>` and `username` with `<username>` so as to be intuitive to the users that these are placeholders.